crypto/src/note.rs

```diff
@@ -81,6 +82,19 @@ impl Note {
         })
     }
 
+    /// Generate a fresh note representing the given value for the given destination address, with a
+    /// random blinding factor.
+    pub fn fresh(
+        rng: &mut impl Rng,
+        address: &crate::Address,
+        value: Value,
+    ) -> Result<Self, Error> {
```

Comment:
Does this need to be fallible? If we're passing in an `Address`, can we be sure that the transmission key it contains is always valid?

Reply:
Yeah, the `Address` is supposed to maintain the invariant that its fields are valid, so the `.expect` can be folded into the body of this function, and the function can be made infallible.